src/main/java/gov/nist/oar/distrib/datapackage/ObjectUtils.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * This software was developed at the National Institute of Standards and Technology by employees of
+ * the Federal Government in the course of their official duties. Pursuant to title 17 Section 105
+ * of the United States Code this software is not subject to copyright protection and is in the
+ * public domain. This is an experimental system. NIST assumes no responsibility whatsoever for its
+ * use by other parties, and makes no guarantees, expressed or implied, about its quality,
+ * reliability, or any other characteristic. We would appreciate acknowledgement if the software is
+ * used. This software can be redistributed and/or modified freely provided that any derivative
+ * works bear some notice that they are derived from it, and any modified versions bear some notice
+ * that they have been modified.
+ * @author: Deoyani Nandrekar-Heinis
+ */
+package gov.nist.oar.distrib.datapackage;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import gov.nist.oar.distrib.web.objects.FileRequest;
+
+/***
+ * ObjectUtils class provides the functionality to validate objects and text.
+ * 
+ * @author Deoyani Nandrekar-Heinis
+ */
+public class ObjectUtils {
+    public ObjectUtils() {
+    }
+
+    /**
+     * Read the url header to get the size of file.
+     * 
+     * @param url
+     * @return long
+     * @throws IOException
+     */
+    public static long getFileSize(String url) throws IOException {
+	URL obj = new URL(url);
+	URLConnection conn = obj.openConnection();
+	return conn.getContentLength();
+    }
+
+    /**
+     * Return true if a given URL matches one of the set of allowed URL patterns.  
+     * 
+     * @param url           the URL to test
+     * @param allowedUrls   a regular expression string for allowed URLs.  Multiple URL patterns 
+     *                      can be concatonated with a pipe (|) character.
+     * @return boolean      True if the URL matches one of the given patterns; false, otherwise
+     * @throws IOException
+     */
+   // private 
+    //private 
+    public static boolean isAllowedURL(String url, String allowedUrls) throws IOException {
```

Comment:
(Re: implementation for `validateUrlDomain()`)

I think this implementation is more complex than it needs to be; it took me a while to convince myself that the implementation does what it's suppose to.  In particular,
  * I'm not sure why it is necessary to split each URL pattern into its authority (machine name) and path.  Wouldn't it be easier to lop off the scheme (e.g. https://) from the URL and match that against each pattern?
  * This only needs to match one pattern to return true, however this implementation always matches against every pattern and then looks to see if one of them matched.  It would be simpler (and more performant) just to return true inside the loop the first time it matches and return false if it exits the loop without doing so.

We should consider going back and tightening this implementation up later.